edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -256,3 +256,45 @@ def test_exams_config(self, other_course_id, exam_type, expect_none_provider):
         self.assertEqual(new_exam.due_date, pytz.utc.localize(datetime.fromisoformat('2025-07-01 00:00:00')))
         self.assertEqual(new_exam.hide_after_due, False)
         self.assertEqual(new_exam.is_active, True)
+
+
+class ProctoringProvidersViewTest(ExamsAPITestCase):
+    """
+    Tests ProctoringProvidersView
+    """
+
+    def get_response(self):
+        """
+        Helper function to make a get request
+        """
+        url = reverse("api:v1:proctoring-providers-list")
+        response = self.client.get(url)
+        return response
+
+    def test_proctoring_providers_list(self):
+        test_provider2 = ProctoringProvider.objects.create(
+            name='test_provider2',
+            verbose_name='testing_provider2',
+            lti_configuration_id='223456789'
+        )
+
+        # import pudb; pu.db
+
+        response = self.get_response()
+        proctoring_providers_list = ProctoringProvider.objects.all()
+
+        expected_data = [self.test_provider.verbose_name, test_provider2.verbose_name]
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(proctoring_providers_list), 2)
+        self.assertIn(response.data[0]['verbose_name'], expected_data)
+        self.assertIn(response.data[1]['verbose_name'], expected_data)
```

Comment:
this would pass if response.data[0] and response.data[1] are the same provider repeated twice by accident

one option is to sort the expected data (maybe call it expected names?) and then sort these two names, and then compare the sorted lists

another option would be to remove the name[0] from the expected data and then check name[1] against the remaining single item list

Reply:
sounds good, with the sort implementation i took an extra step to filter the namessy into a separate list to avoid dealing with OrderedDicts.